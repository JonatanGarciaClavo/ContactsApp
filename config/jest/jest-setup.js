/*  eslint-disable */
// Make Enzyme functions available in all test files without importing
import { shallow, render, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
global.shallow = shallow;
global.render = render;
global.render = mount;
global.shallowToJson = shallowToJson;

// Skip createElement warnings but fail tests on any other warning
console.error = message => {
  if (!/(React.createElement: type should not be null)/.test(message)) {
      throw new Error(message);
  }
};