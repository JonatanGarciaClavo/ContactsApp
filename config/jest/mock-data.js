export const contact = {
  id: '-1',
  email: 'test@test.com',
  imgUrl: 'http://www.test.image.com',
  name: 'Tester',
  phoneNumber: '666666666',
};

export const contacts = [
  {
    id: '-1',
    email: 'test@test.com',
    imgUrl: 'http://www.test.image.com',
    name: 'Tester',
    phoneNumber: '666666666',
  },
  {
    id: '-2',
    email: 'test2@test.com',
    imgUrl: 'http://www.test2.image.com',
    name: 'Tester2',
    phoneNumber: '766666666',
  },
];

export const id = '-1';
