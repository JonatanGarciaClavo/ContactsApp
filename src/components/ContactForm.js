import React, { PropTypes } from 'react';
import { RaisedButton } from 'material-ui';
import TextField from './TextField';

const styles = {
  formContainer: {
    display: 'flex',
    flex: '0 0 auto',
    alignItems: 'center',
    flexDirection: 'column',
  },
  buttonStyle: {
    margin: '2em 0 0 0',
  },
};

class ContactForm extends React.Component {
  constructor(props) {
    super(props);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onSubmit(e) {
    e.preventDefault();
    e.stopPropagation();
    this.props.onSaveClick();
  }

  render() {
    const { contact: { contact: { name, email, phoneNumber, imgUrl }, loading, isModified,
      errors }, onContactAttributeChange, onContactAttributeBlur } = this.props;
    return (
      <div>
        <form style={styles.formContainer} onSubmit={this.onSubmit}>
          <TextField
            label="Name"
            placeholder="Name"
            name="name"
            value={name}
            errorText={errors.name}
            onChange={onContactAttributeChange}
            onBlur={onContactAttributeBlur}
            onEnterKeyDown={this.onSubmit}
          />
          <TextField
            label="Email"
            placeholder="Email"
            name="email"
            value={email}
            errorText={errors.email}
            onChange={onContactAttributeChange}
            onBlur={onContactAttributeBlur}
            onEnterKeyDown={this.onSubmit}
          />
          <TextField
            label="Phone number"
            placeholder="Phone number"
            name="phoneNumber"
            value={phoneNumber}
            errorText={errors.phoneNumber}
            onChange={onContactAttributeChange}
            onBlur={onContactAttributeBlur}
            onEnterKeyDown={this.onSubmit}
          />
          <TextField
            label="Profile image url"
            placeholder="Profile image url"
            name="imgUrl"
            value={imgUrl}
            errorText={errors.imgUrl}
            onChange={onContactAttributeChange}
            onBlur={onContactAttributeBlur}
            onEnterKeyDown={this.onSubmit}
          />
          <RaisedButton
            style={styles.buttonStyle}
            label="Save contact"
            primary
            onTouchEnd={this.onSubmit}
            onClick={this.onSubmit}
            disabled={loading || !isModified}
          />
        </form>
      </div>
    );
  }
}

ContactForm.propTypes = {
  contact: PropTypes.object.isRequired,
  onContactAttributeChange: PropTypes.func.isRequired,
  onContactAttributeBlur: PropTypes.func,
  onSaveClick: PropTypes.func.isRequired,
}

export default ContactForm;
