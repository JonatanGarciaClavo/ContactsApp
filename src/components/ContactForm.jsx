import React, { PropTypes } from 'react';
import { RaisedButton } from 'material-ui';
import { Record } from 'immutable';
import _ from 'lodash';
import TextField from './TextField';

const styles = {
  formContainer: {
    display: 'flex',
    flex: '0 0 auto',
    alignItems: 'center',
    flexDirection: 'column',
  },
  buttonStyle: {
    margin: '2em 0 0 0',
  },
};

class ContactForm extends React.Component {
  constructor(props) {
    super(props);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onSubmit(e) {
    e.preventDefault();
    e.stopPropagation();
    this.props.onSaveClick();
  }

  render() {
    const { contact: { contact: { name, email, phoneNumber, imgUrl }, loading, isModified,
      errors }, onContactAttributeChange, onContactAttributeBlur } = this.props;
    return (
      <div>
        <form style={styles.formContainer} onSubmit={this.onSubmit}>
          <TextField
            label="Name"
            placeholder="Name"
            name="name"
            value={name}
            errorText={errors.get('name')}
            onChange={onContactAttributeChange}
            onBlur={onContactAttributeBlur}
          />
          <TextField
            label="Email"
            placeholder="Email"
            name="email"
            value={email}
            errorText={errors.get('email')}
            onChange={onContactAttributeChange}
            onBlur={onContactAttributeBlur}
          />
          <TextField
            label="Phone number"
            placeholder="Phone number"
            name="phoneNumber"
            value={phoneNumber}
            errorText={errors.get('phoneNumber')}
            onChange={onContactAttributeChange}
            onBlur={onContactAttributeBlur}
          />
          <TextField
            label="Profile image url"
            placeholder="Profile image url"
            name="imgUrl"
            value={imgUrl}
            errorText={errors.get('imgUrl')}
            onChange={onContactAttributeChange}
            onBlur={onContactAttributeBlur}
          />
          <RaisedButton
            style={styles.buttonStyle}
            label="Save contact"
            primary
            onTouchEnd={this.onSubmit}
            onClick={this.onSubmit}
            disabled={loading || !isModified}
          />
        </form>
      </div>
    );
  }
}

ContactForm.defaultProps = {
  onContactAttributeBlur: _.noop,
};

ContactForm.propTypes = {
  contact: PropTypes.instanceOf(Record).isRequired,
  onContactAttributeChange: PropTypes.func.isRequired,
  onContactAttributeBlur: PropTypes.func,
  onSaveClick: PropTypes.func.isRequired,
}

export default ContactForm;
