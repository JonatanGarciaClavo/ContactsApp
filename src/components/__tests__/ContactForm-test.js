import React from 'react';
import renderer from 'react-test-renderer';
import ContactForm from '../ContactForm';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { contact } from '../../../config/jest/mock-data';

// https://github.com/facebook/jest/issues/1353
jest.mock('react/lib/ReactDefaultInjection');

describe('Testing ContactForm component', () => {
  it('renders ContactForm using Snapshots', () => {
    expect(renderer.create(
      <MuiThemeProvider>
        <ContactForm
          contact={{ contact, errors: {} }}
          onContactAttributeChange={jest.fn}
          onContactAttributeBlur={jest.fn}
          onSaveClick={jest.fn}
        />
      </MuiThemeProvider>
    )).toMatchSnapshot();
  });
});
