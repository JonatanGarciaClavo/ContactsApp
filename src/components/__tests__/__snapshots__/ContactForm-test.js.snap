exports[`Testing ContactForm component renders ContactForm using Snapshots 1`] = `
<div>
  <form
    onSubmit={[Function]}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flex": "0 0 auto",
        "flexDirection": "column",
      }
    }>
    <div
      className={undefined}
      style={
        Object {
          "backgroundColor": "transparent",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 16,
          "height": 72,
          "lineHeight": "24px",
          "muiPrepared": true,
          "position": "relative",
          "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": 256,
        }
      }>
      <label
        className={undefined}
        htmlFor="undefined-Name-Name-49349"
        onTouchTap={undefined}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.5)",
            "cursor": "text",
            "lineHeight": "22px",
            "muiPrepared": true,
            "pointerEvents": "none",
            "position": "absolute",
            "top": 38,
            "transform": "scale(0.75) translate(0, -28px)",
            "transformOrigin": "left top",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "zIndex": 1,
          }
        }>
        Name
      </label>
      <div
        style={
          Object {
            "bottom": 12,
            "color": "rgba(0, 0, 0, 0.3)",
            "muiPrepared": true,
            "opacity": 0,
            "position": "absolute",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }>
        Name
      </div>
      <input
        disabled={false}
        id="undefined-Name-Name-49349"
        onBlur={[Function]}
        onChange={[Function]}
        onEnterKeyDown={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "appearance": "textfield",
            "backgroundColor": "rgba(0,0,0,0)",
            "border": "none",
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "initial",
            "font": "inherit",
            "height": "100%",
            "marginTop": 14,
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "width": "100%",
          }
        }
        type="text"
        value="Tester" />
      <div>
        <hr
          style={
            Object {
              "border": "none",
              "borderBottom": "solid 1px",
              "borderColor": "#e0e0e0",
              "bottom": 8,
              "boxSizing": "content-box",
              "margin": 0,
              "muiPrepared": true,
              "position": "absolute",
              "width": "100%",
            }
          } />
        <hr
          style={
            Object {
              "border": "none",
              "borderBottom": "solid 2px",
              "borderColor": "#00bcd4",
              "bottom": 8,
              "boxSizing": "content-box",
              "margin": 0,
              "muiPrepared": true,
              "position": "absolute",
              "transform": "scaleX(0)",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": "100%",
            }
          } />
      </div>
    </div>
    <div
      className={undefined}
      style={
        Object {
          "backgroundColor": "transparent",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 16,
          "height": 72,
          "lineHeight": "24px",
          "muiPrepared": true,
          "position": "relative",
          "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": 256,
        }
      }>
      <label
        className={undefined}
        htmlFor="undefined-Email-Email-24049"
        onTouchTap={undefined}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.5)",
            "cursor": "text",
            "lineHeight": "22px",
            "muiPrepared": true,
            "pointerEvents": "none",
            "position": "absolute",
            "top": 38,
            "transform": "scale(0.75) translate(0, -28px)",
            "transformOrigin": "left top",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "zIndex": 1,
          }
        }>
        Email
      </label>
      <div
        style={
          Object {
            "bottom": 12,
            "color": "rgba(0, 0, 0, 0.3)",
            "muiPrepared": true,
            "opacity": 0,
            "position": "absolute",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }>
        Email
      </div>
      <input
        disabled={false}
        id="undefined-Email-Email-24049"
        onBlur={[Function]}
        onChange={[Function]}
        onEnterKeyDown={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "appearance": "textfield",
            "backgroundColor": "rgba(0,0,0,0)",
            "border": "none",
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "initial",
            "font": "inherit",
            "height": "100%",
            "marginTop": 14,
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "width": "100%",
          }
        }
        type="text"
        value="test@test.com" />
      <div>
        <hr
          style={
            Object {
              "border": "none",
              "borderBottom": "solid 1px",
              "borderColor": "#e0e0e0",
              "bottom": 8,
              "boxSizing": "content-box",
              "margin": 0,
              "muiPrepared": true,
              "position": "absolute",
              "width": "100%",
            }
          } />
        <hr
          style={
            Object {
              "border": "none",
              "borderBottom": "solid 2px",
              "borderColor": "#00bcd4",
              "bottom": 8,
              "boxSizing": "content-box",
              "margin": 0,
              "muiPrepared": true,
              "position": "absolute",
              "transform": "scaleX(0)",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": "100%",
            }
          } />
      </div>
    </div>
    <div
      className={undefined}
      style={
        Object {
          "backgroundColor": "transparent",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 16,
          "height": 72,
          "lineHeight": "24px",
          "muiPrepared": true,
          "position": "relative",
          "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": 256,
        }
      }>
      <label
        className={undefined}
        htmlFor="undefined-Phonenumber-Phonenumber-39637"
        onTouchTap={undefined}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.5)",
            "cursor": "text",
            "lineHeight": "22px",
            "muiPrepared": true,
            "pointerEvents": "none",
            "position": "absolute",
            "top": 38,
            "transform": "scale(0.75) translate(0, -28px)",
            "transformOrigin": "left top",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "zIndex": 1,
          }
        }>
        Phone number
      </label>
      <div
        style={
          Object {
            "bottom": 12,
            "color": "rgba(0, 0, 0, 0.3)",
            "muiPrepared": true,
            "opacity": 0,
            "position": "absolute",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }>
        Phone number
      </div>
      <input
        disabled={false}
        id="undefined-Phonenumber-Phonenumber-39637"
        onBlur={[Function]}
        onChange={[Function]}
        onEnterKeyDown={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "appearance": "textfield",
            "backgroundColor": "rgba(0,0,0,0)",
            "border": "none",
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "initial",
            "font": "inherit",
            "height": "100%",
            "marginTop": 14,
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "width": "100%",
          }
        }
        type="text"
        value="666666666" />
      <div>
        <hr
          style={
            Object {
              "border": "none",
              "borderBottom": "solid 1px",
              "borderColor": "#e0e0e0",
              "bottom": 8,
              "boxSizing": "content-box",
              "margin": 0,
              "muiPrepared": true,
              "position": "absolute",
              "width": "100%",
            }
          } />
        <hr
          style={
            Object {
              "border": "none",
              "borderBottom": "solid 2px",
              "borderColor": "#00bcd4",
              "bottom": 8,
              "boxSizing": "content-box",
              "margin": 0,
              "muiPrepared": true,
              "position": "absolute",
              "transform": "scaleX(0)",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": "100%",
            }
          } />
      </div>
    </div>
    <div
      className={undefined}
      style={
        Object {
          "backgroundColor": "transparent",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 16,
          "height": 72,
          "lineHeight": "24px",
          "muiPrepared": true,
          "position": "relative",
          "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": 256,
        }
      }>
      <label
        className={undefined}
        htmlFor="undefined-Profileimageurl-Profileimageurl-3097"
        onTouchTap={undefined}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.5)",
            "cursor": "text",
            "lineHeight": "22px",
            "muiPrepared": true,
            "pointerEvents": "none",
            "position": "absolute",
            "top": 38,
            "transform": "scale(0.75) translate(0, -28px)",
            "transformOrigin": "left top",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "zIndex": 1,
          }
        }>
        Profile image url
      </label>
      <div
        style={
          Object {
            "bottom": 12,
            "color": "rgba(0, 0, 0, 0.3)",
            "muiPrepared": true,
            "opacity": 0,
            "position": "absolute",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }>
        Profile image url
      </div>
      <input
        disabled={false}
        id="undefined-Profileimageurl-Profileimageurl-3097"
        onBlur={[Function]}
        onChange={[Function]}
        onEnterKeyDown={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "appearance": "textfield",
            "backgroundColor": "rgba(0,0,0,0)",
            "border": "none",
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "initial",
            "font": "inherit",
            "height": "100%",
            "marginTop": 14,
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "width": "100%",
          }
        }
        type="text"
        value="http://www.test.image.com" />
      <div>
        <hr
          style={
            Object {
              "border": "none",
              "borderBottom": "solid 1px",
              "borderColor": "#e0e0e0",
              "bottom": 8,
              "boxSizing": "content-box",
              "margin": 0,
              "muiPrepared": true,
              "position": "absolute",
              "width": "100%",
            }
          } />
        <hr
          style={
            Object {
              "border": "none",
              "borderBottom": "solid 2px",
              "borderColor": "#00bcd4",
              "bottom": 8,
              "boxSizing": "content-box",
              "margin": 0,
              "muiPrepared": true,
              "position": "absolute",
              "transform": "scaleX(0)",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": "100%",
            }
          } />
      </div>
    </div>
    <div
      className={undefined}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": "2px",
          "boxShadow": undefined,
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "margin": "2em 0 0 0",
          "minWidth": 88,
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        }
      }>
      <button
        disabled={true}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "rgb(229, 229, 229)",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "cursor": "default",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transform": "translate(0, 0)",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
            "width": "100%",
          }
        }
        tabIndex={0}
        type="button">
        <div
          style={
            Object {
              "backgroundColor": false,
              "borderRadius": 2,
              "height": 36,
              "muiPrepared": true,
              "top": 0,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }>
          <span
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.3)",
                "fontSize": 14,
                "fontWeight": 500,
                "letterSpacing": 0,
                "margin": 0,
                "muiPrepared": true,
                "opacity": 1,
                "paddingLeft": 16,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "userSelect": "none",
              }
            }>
            Save contact
          </span>
        </div>
      </button>
    </div>
  </form>
</div>
`;
